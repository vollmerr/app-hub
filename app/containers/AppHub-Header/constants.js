/*
 *
 * AppHubHeader constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHeader/DEFAULT_ACTION';
