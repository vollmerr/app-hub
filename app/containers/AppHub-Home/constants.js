/*
 *
 * AppHubHome constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHome/DEFAULT_ACTION';
