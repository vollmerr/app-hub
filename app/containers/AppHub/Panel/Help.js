import React from 'react';

export default () => <div>in Help panel</div>;
