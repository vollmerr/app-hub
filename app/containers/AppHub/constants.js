// VIEW
export const CHANGE_MOBILE = 'AppHub/CHANGE_MOBILE';
export const CHANGE_PANEL_OPEN = 'AppHub/CHANGE_PANEL_OPEN';
export const CHANGE_PANEL_SELECTED = 'AppHub/CHANGE_PANEL_SELECTED';
// panel names
export const HELP_PANEL = 'AppHub/HELP_PANEL';
export const APPS_PANEL = 'AppHub/APPS_PANEL';
export const APP_NAV_PANEL = 'AppHub/APP_NAV_PANEL';
export const ALERTS_PANEL = 'AppHub/ALERTS_PANEL';
export const DEV_PANEL = 'AppHub/DEV_PANEL';
// APP
export const CHANGE_APP = 'AppHub/CHANGE_APP';
export const CHANGE_APP_STATUS = 'AppHub/CHANGE_APP_STATUS';
// user
export const AUTH_USER = 'AppHub/AUTH_USER';
export const AUTH_USER_DONE = 'AppHub/AUTH_USER_DONE';
// REQUEST STATUS TYPES
export const REQUEST = '_REQUEST';
export const SUCCESS = '_SUCCESS';
export const FAILURE = '_FAILURE';

export const COL_TYPES = {
  DATE: 'DATE',
};

export const ROLES = {
  USER: 'AppHub User',
};

// D3
export const INCREMENT_RENDER_COUNT = 'AppHub/INCREMENT_RENDER_COUNT';
