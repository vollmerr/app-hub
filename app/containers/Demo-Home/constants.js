export const EXAMPLE_DATA_REQUEST = 'app/DemoHome/EXAMPLE_DATA_REQUEST';
export const EXAMPLE_DATA_SUCCESS = 'app/DemoHome/EXAMPLE_DATA_SUCCESS';
export const EXAMPLE_DATA_FAILURE = 'app/DemoHome/EXAMPLE_DATA_FAILURE';
