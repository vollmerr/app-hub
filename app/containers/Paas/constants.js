export const GET_MANAGER_DATA_REQUEST = 'app/Paas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Paas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Paas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Paas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Paas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Paas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Paas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Paas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Paas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Paas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Paas/SET_REPORT_FILTER';

// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSid',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};
