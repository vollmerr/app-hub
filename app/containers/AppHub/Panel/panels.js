import * as C from '../constants';

import AppNav from '../../App/Nav';

import Help from './Help';
import AppTiles from './AppTiles';
import Alerts from './Alerts';
import Dev from './Dev';
// import Mock from './Mock';


export default {
  [C.HELP_PANEL]: {
    component: Help,
  },
  [C.APPS_PANEL]: {
    component: AppTiles,
  },
  [C.APP_NAV_PANEL]: {
    isLeft: true,
    component: AppNav,
  },
  [C.ALERTS_PANEL]: {
    component: Alerts,
  },
  [C.DEV_PANEL]: {
    component: Dev,
  },
  // [C.MOCK_PANEL]: {
  //   component: Mock,
  // },
};
