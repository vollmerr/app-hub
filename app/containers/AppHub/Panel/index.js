import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import toJS from '../../../hocs/toJS';
import Wrapper from '../../../components/Panel';

import * as selectors from '../selectors';

import panels from './panels';


class Panel extends React.PureComponent {
  render() {
    const { app, user, view, panel, onClick } = this.props;
    const Content = panels[panel.name].component;

    const panelProps = {
      onClick,
      isOpen: panel.isOpen,
      isLeft: panels[panel.name].isLeft,
    };

    const contentProps = {
      appRotues: app.routes,
      userRoutes: user.routes,
      isMobile: view.isMobile,
      onClick: () => onClick(null),
    };

    /* istanbul ignore next: TODO: how to test dynamic built components */
    return (
      <Wrapper {...panelProps}>
        <Content {...contentProps} />
      </Wrapper>
    );
  }
}


const { func, object } = PropTypes;

Panel.propTypes = {
  app: object.isRequired,
  user: object.isRequired,
  view: object.isRequired,
  panel: object.isRequired,
  onClick: func.isRequired,
};


const mapStateToProps = createStructuredSelector({
  app: selectors.getApp,
  user: selectors.getUser,
  view: selectors.getView,
  panel: selectors.getViewPanel,
});

const withConnect = connect(mapStateToProps);


export default compose(
  withConnect,
  toJS,
)(Panel);
