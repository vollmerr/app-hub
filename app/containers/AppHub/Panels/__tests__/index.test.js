import React from 'react';
import { shallow } from 'enzyme';

import * as C from '../../constants';

import Panel from '../../../../components/Panel';

import { Panels } from '../index';
import panels from '../panels';


const props = {
  user: {
    routes: [{ key: 'rotue1', name: 'route1', path: '/test' }],
  },
  view: {
    isMobile: false,
  },
  panel: {
    name: C.HELP_PANEL,
    isOpen: true,
  },
  onClick: jest.fn(),
};


describe('<Panels />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Panels {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a `Panel`', () => {
    expect(wrapper.find(Panel).length).toEqual(1);
  });

  it('should render the correct panel based off `panel`', () => {
    Object.keys(panels).forEach((panel) => {
      wrapper.setProps({ panel: { name: panel, isOpen: true } });
      expect(wrapper.find(panels[panel].component).length).toEqual(1);
    });

    expect(wrapper.children().length).toEqual(1);
  });

  it('should call the `onClick` with nothing when clicking on content (not a panel)', () => {
    wrapper.find(panels[props.panel.name].component).simulate('click');
    expect(props.onClick).toHaveBeenCalledWith(null);
  });
});
