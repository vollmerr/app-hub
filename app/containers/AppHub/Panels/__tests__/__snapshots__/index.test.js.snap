// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panels /> should render correctly 1`] = `
<Panel
  isOpen={true}
  onClick={[Function]}
>
  <Help
    isMobile={false}
    onClick={[Function]}
    routes={
      Array [
        Object {
          "key": "rotue1",
          "name": "route1",
          "path": "/test",
        },
      ]
    }
  />
</Panel>
`;
